great_expectations/data_context/data_context/abstract_data_context.py

```diff
@@ -743,12 +745,13 @@ def _add_fluent_datasource(
         # temporary workaround while we update stores to work better with Fluent Datasources for all contexts
         # Without this we end up with duplicate entries for datasources in both
         # "fluent_datasources" and "datasources" config/yaml entries.
-        if self._datasource_store.cloud_mode:
+        if self._datasource_store.cloud_mode and not from_config:
```

Comment:
We need to add this branch here otherwise the initial `get_context()` call will result in `PUT` update requests for every fluent datasource that was just pulled from cloud.

I will follow this up with a refactor PR to breakup this method up into 2 smaller methods and remove this extra condition.